### PR TITLE
SPORTS-21 Initial project setup for front-end application and folder structure

### PR DESCRIPTION
**Pull request by sagunji**
- Use Webpack
- No Typescript (we can add it later if we wish to but not compulsion)

### CODE REVIEW

File: chess/frontend/public/index.html [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+
+    <title>LF-SPORTS</title>
+  </head>
+  <body>
+    <div id="app"></div>
+  </body>
+</html>
+
```
Comment (cham11ng) [inline chess/frontend/public/index.html]: .

File: chess/frontend/src/index.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
```
Comment (sanjeevkpandit) [inline chess/frontend/src/index.js]: Length rule

File: chess/frontend/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
+
+const path = require('path');
+
```
Comment (sanjeevkpandit) [inline chess/frontend/webpack.config.js]: Unnecessary new lines

File: chess/frontend/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
```
Comment (sanjeevkpandit) [inline chess/frontend/webpack.config.js]: Unnecessary new lines

File: chess/frontend/package.json [hunk 5]
```diff
@@ -4,8 +4,28 @@
...
+    "babel-loader": "^7.1.2",
+    "babel-preset-env": "^1.6.1",
+    "babel-preset-react": "^6.24.1",
+    "css-loader": "^0.28.9",
+    "extract-text-webpack-plugin": "^3.0.2",
+    "file-loader": "^1.1.6",
+    "html-webpack-plugin": "^2.30.1",
+    "webpack": "^3.11.0",
+    "webpack-dev-server": "^2.11.1"
+  },
```
Comment (cham11ng) [inline chess/frontend/package.json]: I think we also require `"node-sass", "sass-loader", "style-loader"`  these plugins for SASS pre-processing

File: chess/frontend/webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
```
Comment (cham11ng) [inline chess/frontend/webpack.config.js]: .

File: chess/frontend/public/index.html [hunk 7]
```diff
@@ -0,0 +1,13 @@
+<!doctype html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width">
+
```
Comment (aviskarkc10) [inline chess/frontend/public/index.html]: let's remove this line as well

File: chess/frontend/webpack.config.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          use: 'css-loader',
+        }),
+      },
+      {
+        test: /\.(png|jpg|gif)$/,
+        use: [
+          'file-loader',
+        ],
+      },
+    ],
```
Comment (cham11ng) [inline chess/frontend/webpack.config.js]: I used this before SASS processing rule
```js
{
  test: /\.s[ac]ss$/,
  use: ExtractTextPlugin.extract({
    use: ['css-loader', 'sass-loader'],
      fallback: 'style-loader'
  })
}, 
```

File: chess/frontend/public/index.html (hunk 7 above)
Comment (sagunji) [reply chess/frontend/public/index.html]: meta tag?

Comment (sanjeevkpandit) [reply chess/frontend/public/index.html]: @sagunji  the new line

File: chess/frontend/public/index.html (hunk 1 above)
Comment (sagunji) [reply chess/frontend/public/index.html]: New line?

Comment (cham11ng) [reply chess/frontend/public/index.html]: I think there is unnecessary space

Comment (sagunji) [reply chess/frontend/public/index.html]: its alright, I added a new line at the end of every file.

**Pull request merged**